### ISSUE TITLE
Lua 5.3 support for Fedora 22+

### ISSUE THREAD

**Issue by ryzom-pipeline** [ryzom/ryzomcore#262]
**[Original report](https://bitbucket.org/ryzom/ryzomcore/issue/262) by Erwan GEORGET (Bitbucket: [Dremor56](https://bitbucket.org/Dremor56), ).**

----------------------------------------

All current Fedora version (22 and 23) now use Lua 5.3. Thus,  /code/CMakeModules/FindLuabind.cmake need to be changed to reflect these changes.

**Comment by ryzom-pipeline** [comment]
**Original comment by Cédric Ochs (Bitbucket: [Cédric OCHS](https://bitbucket.org/Cédric OCHS), ).**

----------------------------------------

Hi @Dremor56 :)

Thanks for your report, please what changes do we need to do for Lua 5.3 being correctly detected ?

Because, we're currently using Lua 5.3 under all platforms and we fixed the new bugs we discovered.

**Comment by ryzom-pipeline** [comment]
**Original comment by Erwan GEORGET (Bitbucket: [Dremor56](https://bitbucket.org/Dremor56), ).**

----------------------------------------

If I'm not mistaken ,
```
#!cmake

IF(NOT LUALIB_FOUND)
      # fedora (v20)
      SET(LUA52_LIBRARY "liblua-5.2")
      CHECK_LINKED_LIBRARY(LUABIND_LIBRARY_RELEASE LUA52_LIBRARY LUALIB_FOUND)
    ENDIF(NOT LUALIB_FOUND)
```
 Should be changed to :

```
#!cmake

IF(NOT LUALIB_FOUND)
      # fedora (v22+)
      SET(LUA53_LIBRARY "liblua-5.3")
      CHECK_LINKED_LIBRARY(LUABIND_LIBRARY_RELEASE LUA53_LIBRARY LUALIB_FOUND)
    ENDIF(NOT LUALIB_FOUND)
```
Note : I'm a beginner with cmake (helping the Khaganat team since yesterday), this code should probably be reviewed by someone who know it better than me.

**Comment by ryzom-pipeline** [comment]
**Original comment by Cédric Ochs (Bitbucket: [Cédric OCHS](https://bitbucket.org/Cédric OCHS), ).**

----------------------------------------

Don't worry :) Your proposition explains exactly what changed, that's exactly what I wanted :) Thanks !

I'll make a fix soon :)

**Comment by ryzom-pipeline** [comment]
**Original comment by Cédric Ochs (Bitbucket: [Cédric OCHS](https://bitbucket.org/Cédric OCHS), ).**

----------------------------------------

Lua 5.3 support for Fedora 22+, fixes #262

**Comment by ryzom-pipeline** [comment]
**Original comment by Cédric Ochs (Bitbucket: [Cédric OCHS](https://bitbucket.org/Cédric OCHS), ).**

----------------------------------------

Please could you confirm if it's working now ?

Thanks :)

**Comment by ryzom-pipeline** [comment]
**Original comment by Erwan GEORGET (Bitbucket: [Dremor56](https://bitbucket.org/Dremor56), ).**

----------------------------------------

I'll try tomorrow.

**Comment by ryzom-pipeline** [comment]
**Original comment by Erwan GEORGET (Bitbucket: [Dremor56](https://bitbucket.org/Dremor56), ).**

----------------------------------------

"Can't find Lua version used by Luabind"

**Comment by ryzom-pipeline** [comment]
**Original comment by Cédric Ochs (Bitbucket: [Cédric OCHS](https://bitbucket.org/Cédric OCHS), ).**

----------------------------------------

Improved Luabind CMake module, fixes #262 and fixes #263

**Comment by ryzom-pipeline** [comment]
**Original comment by Erwan GEORGET (Bitbucket: [Dremor56](https://bitbucket.org/Dremor56), ).**

----------------------------------------

Feature good to go, cmake complete configuration under Fedora without problems.
